File: TCP/tcp_server.py
import socket

def start_server():
    host = '127.1.0.1'
    port = 65446

    server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    server_socket.bind((host, port))
    server_socket.listen(1)

    print(f"TCP Server Connecting on {host}:{port}...")

    conn, addr = server_socket.accept()
    print(f"Connected by {addr}")

    while True:
        try:
            data = conn.recv(1024)
            if not data:
                print("Client disconnected.")
                break
            print("Client says:", data.decode())

            reply = input("Enter reply to Client: ")
            conn.sendall(reply.encode())
        except ConnectionResetError:
            print("Client closed the connection.")
            break

    conn.close()
    server_socket.close()

if __name__ == "__main__":
    start_server()

File: Java/TCPServer.java
import java.io.*;
import java.net.*;

public class TCPServer {
    public static void main(String[] args) {
        String host = "127.0.0.1";  // localhost IP
        int port = 65446;

        try (ServerSocket serverSocket = new ServerSocket()) {
            serverSocket.bind(new InetSocketAddress(host, port));
            System.out.println("TCP Server connecting on " + host + ":" + port + "...");

            Socket clientSocket = serverSocket.accept();
            System.out.println("Connected by " + clientSocket.getRemoteSocketAddress());

            BufferedReader in = new BufferedReader(new InputStreamReader(clientSocket.getInputStream()));
            BufferedWriter out = new BufferedWriter(new OutputStreamWriter(clientSocket.getOutputStream()));
            BufferedReader stdIn = new BufferedReader(new InputStreamReader(System.in));

            String clientMsg;
            while ((clientMsg = in.readLine()) != null) {
                System.out.println("Client says: " + clientMsg);

                System.out.print("Enter reply to Client: ");
                String reply = stdIn.readLine();
                if (reply == null) break;

                out.write(reply);
                out.newLine();
                out.flush();
            }

            System.out.println("Client disconnected.");
            clientSocket.close();

        } catch (IOException e) {
            System.err.println("Server error: " + e.getMessage());
        }
    }
}
